coursier.bzl

```diff
@@ -261,6 +261,50 @@ def get_home_netrc_contents(repository_ctx):
                 return repository_ctx.read(netrcfile)
     return ""
 
+def generate_coursier_credentials_from_netrc(netrc_content):
+    """Generate a list of coursier credentials from a netrc file. Only
+    'machine' entries containing a login and a password are taken into
+    account. The 'default' entry is ignored.
+    """
+    spacified = netrc_content.replace("\n", " ").replace("\t", " ")
+    sanitized = [e for e in spacified.split(" ") if e != ""]
+
+    machines = {}
+    machine = None
+    i = 0
+    for _ in sanitized:
+        if i >= len(sanitized):
+            break
+        token = sanitized[i]
+        if token == "default":
+            machine = None
+            i = i + 1
+        elif token == "machine":
+            if i + 1 > len(sanitized):
+                fail("netrc file is not correct (error on token {})".format(token))
```

Comment:
Should this fail the whole run or just print a warning and continue without any credentials set?

Reply:
Would be nice if netrc files with non-standard syntax didn't fail the build.

Reply:
I changed the function to use the `read_netrc` helper provided by Bazel.